fe/fe-core/src/main/java/com/starrocks/alter/SystemHandler.java

```diff
@@ -221,11 +227,39 @@ public static List<Backend> checkDecommission(List<Pair<String, Integer>> hostPo
                 // already under decommission, ignore it
                 continue;
             }
+            needCapacity += backend.getDataUsedCapacityB();
+            releaseCapacity += backend.getAvailableCapacityB();
             decommissionBackends.add(backend);
         }
 
-        // TODO(cmy): check if replication num can be met
-        // TODO(cmy): check remaining space
+        short maxReplicationNum = 0;
+        LocalMetastore localMetastore = GlobalStateMgr.getCurrentState().getLocalMetastore();
+        for (long dbId : localMetastore.getDbIds()) {
+            Database db = localMetastore.getDb(dbId);
+            for (Table table : db.getTables()) {
+                if (table instanceof OlapTable) {
+                    OlapTable olapTable = (OlapTable) table;
+                    PartitionInfo partitionInfo = olapTable.getPartitionInfo();
+                    for (long partitionId : olapTable.getAllPartitionIds()) {
+                        short replicationNum = partitionInfo.getReplicationNum(partitionId);
+                        if (replicationNum > maxReplicationNum) {
+                            maxReplicationNum = replicationNum;
+                            if (infoService.getAliveBackendNumber() - decommissionBackends.size() < maxReplicationNum) {
+                                decommissionBackends.clear();
+                                throw new DdlException("It will cause insufficient BE number if these BEs are decommissioned " +
+                                        "because the table " + db.getFullName() + "." + olapTable.getName() + " requires " +
+                                        maxReplicationNum + " replication number.");
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        if (infoService.getClusterAvailableCapacityB() - releaseCapacity < needCapacity) {
```

Comment:
Plz try to add ut for these codes, because 80% ut coverage ratio is a requirement for pr to merge now.